pkg/limayaml/validate.go

```diff
@@ -641,3 +655,36 @@ func ValidateAgainstLatestConfig(yNew, yLatest []byte) error {
 
 	return nil
 }
+
+func validateSocket(engine, socket string) error {
+	if socket == "" {
+		return fmt.Errorf("%s socket path must not be empty", engine)
+	}
+
+	u, err := url.Parse(socket)
+	if err == nil && u.Scheme != "" {
+		switch u.Scheme {
+		case "unix", "file":
+			if u.Path == "" {
+				return fmt.Errorf("%s socket path %q is not a valid URL: missing path", engine, socket)
+			}
+			return validateUnixSocket(engine, u.Path)
+		case "tcp":
+			if u.Host == "" {
+				return fmt.Errorf("%s socket path %q is not a valid URL: missing host", engine, socket)
+			}
+			return nil
+		default:
+			return fmt.Errorf("%s socket path %q is not a valid URL: unsupported scheme %q", engine, socket, u.Scheme)
+		}
+	}
+
+	return nil
+}
+
+func validateUnixSocket(engine, path string) error {
+	if !filepath.IsAbs(path) {
+		return fmt.Errorf("%s socket path must be absolute, got %s", engine, path)
```

Comment:
Use default validation error format. Always use `%q` for user-supplied strings.

```suggestion
func validateUnixSocket(field, path string) error {
	if !filepath.IsAbs(path) {
		return fmt.Errorf("field `%s` must be absolute, got %q", engine, path)
```